src/action/main.ts

```diff
@@ -51,39 +51,41 @@ async function run(): Promise<void> {
-      argument('--forge-api-host', inputs.github.apiUrl),
-      argument('--ignore-opts-files', inputs.steward.ignoreOptsFiles),
-      argument('--sign-commits', inputs.commits.sign.enabled),
-      argument('--git-author-signing-key', inputs.commits.sign.key),
-      argument('--cache-ttl', inputs.steward.cacheTtl),
-      argument('--max-buffer-size', inputs.steward.maxBufferSize),
-      argument('--scalafix-migrations', inputs.migrations.scalafix),
-      argument('--artifact-migrations', inputs.migrations.artifacts),
-      argument('--repo-config', inputs.steward.defaultConfiguration),
-      argument('--github-app-id', inputs.github.app && !inputs.github.app.authOnly ? inputs.github.app.id : undefined),
-      argument('--github-app-key-file', inputs.github.app && !inputs.github.app.authOnly ? workspace.app_pem : undefined),
-      '--do-not-fork',
-      '--disable-sandbox',
-      inputs.steward.extraArgs?.value.split(' ') ?? [],
-    ], inputs.steward.extraJars)
+    try {
+      await coursier.launch(app, [
+        argument('--workspace', workspace.workspace),
+        argument('--repos-file', workspace.repos_md),
+        argument('--git-ask-pass', workspace.askpass_sh),
+        argument('--git-author-email', inputs.commits.author.email ?? user.email()),
+        argument('--git-author-name', inputs.commits.author.name ?? user.name()),
+        argument('--forge-login', user.login()),
+        argument('--env-var', nonEmpty('"SBT_OPTS=-Xmx2048m -Xss8m -XX:MaxMetaspaceSize=512m"')),
+        argument('--process-timeout', inputs.steward.timeout),
+        argument('--forge-api-host', inputs.github.apiUrl),
+        argument('--ignore-opts-files', inputs.steward.ignoreOptsFiles),
+        argument('--sign-commits', inputs.commits.sign.enabled),
+        argument('--git-author-signing-key', inputs.commits.sign.key),
+        argument('--cache-ttl', inputs.steward.cacheTtl),
+        argument('--max-buffer-size', inputs.steward.maxBufferSize),
+        argument('--scalafix-migrations', inputs.migrations.scalafix),
+        argument('--artifact-migrations', inputs.migrations.artifacts),
+        argument('--repo-config', inputs.steward.defaultConfiguration),
+        argument('--github-app-id', inputs.github.app && !inputs.github.app.authOnly ? inputs.github.app.id : undefined),
+        argument('--github-app-key-file', inputs.github.app && !inputs.github.app.authOnly ? workspace.app_pem : undefined),
+        '--do-not-fork',
+        '--disable-sandbox',
+        inputs.steward.extraArgs?.value.split(' ') ?? [],
+      ], inputs.steward.extraJars)
+    } finally {
+      await workspace.saveWorkspaceCache()
+    }
 
     if (files.existsSync(workspace.runSummary_md)) {
       logger.info(`✓ Run Summary file: ${workspace.runSummary_md}`)
 
       const summaryMarkdown = files.readFileSync(workspace.runSummary_md, 'utf8')
       await core.summary.addRaw(summaryMarkdown).write()
     }
-
-    await workspace.saveWorkspaceCache()
```

Comment:
Oh, small problem here! The `saveWorkspaceCache()` method **deletes** temporary files (like `run-summary.md`) before saving the cache, and moving `saveWorkspaceCache()` to _before_ the run-summary processing-code unfortunately means that the `run-summary.md` is deleted before it's even read..!

See this PR for the fix:

* https://github.com/scala-steward-org/scala-steward-action/pull/652